### PR TITLE
[WIP] feat: New Form.Item validate design

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

Group 24

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     New Form.Item validation style design. Provides `feedback` & `feedbackIcon` for customize validation. Deprecated `help` & `hasFeedback`.      |
| 🇨🇳 Chinese |      全新的 Form.Item 校验样式设计。新增 `feedback` & `feedbackIcon`属性用于自定义校验。 原 `help` & `hasFeedback` 属性废弃。    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/form/demo/disabled-input-debug.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/demo/disabled-input-debug.md)
[View rendered components/form/demo/register.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/demo/register.md)
[View rendered components/form/demo/validate-other.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/demo/validate-other.md)
[View rendered components/form/demo/validate-static-legacy.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/demo/validate-static-legacy.md)
[View rendered components/form/demo/validate-static.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/demo/validate-static.md)
[View rendered components/form/demo/without-form-create.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/demo/without-form-create.md)
[View rendered components/form/index.en-US.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/index.en-US.md)
[View rendered components/form/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/form-status/components/form/index.zh-CN.md)

### CODE REVIEW

File: components/form/ErrorList.tsx [hunk 1]
```diff
@@ -1,70 +1,88 @@
...
-  );
+  const [feedbackVisible, feedbackList, feedbackStatus] = useStatusFeedback(errors, validateStatus);
 
-  // Memo status in same visible
-  const [innerStatus, setInnerStatus] = React.useState(status);
-  React.useEffect(() => {
-    if (visible && status) {
-      setInnerStatus(status);
+  useIsomorphicEffect(() => {
+    if (feedbackVisible) {
```
Comment (hengkx) [inline components/form/ErrorList.tsx:70]: 使用这个 是为了 防止 页面闪烁吗？

Comment (zombieJ) [reply components/form/ErrorList.tsx:70]: 嗯

File: components/form/index.en-US.md [hunk 2]
```diff
@@ -75,10 +75,10 @@ Form field component for data bidirectional binding, validation, layout, and so
 | colon | Used with `label`, whether to display `:` after label text. | boolean | true |  |
 | dependencies | Set the dependency field. See [below](#dependencies) | [NamePath](#NamePath)[] | - |  |
 | extra | The extra prompt message. It is similar to help. Usage example: to display error message and prompt message at the same time | string \| ReactNode | - |  |
+| feedback | The prompt message. If not provided, the prompt message will be generated by the validation rule. | string \| ReactNode | - | 4.7.0 |
```
Comment (afc163) [inline components/form/index.en-US.md:78]: 描述里可以提一下 help，用的比较多。

**Comment by afc163** [comment]
image

没对齐。

**Comment by afc163** [comment]
image

说实话，这个默认效果有点太重了，尤其是**实心的**红色错误 icon。

**Comment by afc163** [comment]
对设计来说算是 break changing。

Form 上的 feedbackIcon 也要提供，让用户可以统一关掉，包括 ConfigProvider。

**Comment by zombieJ** [comment]
先改成 WIP，和设计师再同步一下

**Comment by AlanCutFlim** [comment]
请问这个 PR 现在的进度怎么样了？

**Comment by yuxuan** [comment]
当有extra的时候，这个会怎么展示呢？

**Pull request closed without merging**